tests/README.md

```diff
@@ -0,0 +1,515 @@
+# Getting Started with Consensus Spec Tests
+
+## Getting Started
+
+### Creating the environment
+
+Use an OS that has Python 3.8 or above. For example, Debian 11 (bullseye)
+
+1. Install the packages you need:
+   ```sh
+   sudo apt install -y make git wget python3-venv gcc python3-dev
+   ```
+1. Download the latest [consensus specs](https://github.com/ethereum/consensus-specs)
+   ```sh
+   git clone https://github.com/ethereum/consensus-specs.git
+   cd consensus-specs
+   ```
+1. Create the specifications and tests:   
+   ```sh
+   make install_test
+   make pyspec
+   ```
+
+### Running your first test
+
+
+1. Enter the virtual Python environment:
+   ```sh
+   cd ~/consensus-specs
+   . venv/bin/activate
+   ```
+1. Run a sanity check test:
+   ```sh 
+   cd tests/core/pyspec
+   python -m pytest -k test_empty_block_transition --fork Merge eth2spec
+   ```
+1. The output should be similar to:
+   ```
+   ============================= test session starts ==============================
+   platform linux -- Python 3.9.2, pytest-6.2.5, py-1.10.0, pluggy-1.0.0
+   rootdir: /home/qbzzt1/consensus-specs
+   plugins: cov-2.12.1, forked-1.3.0, xdist-2.3.0
+   collected 629 items / 626 deselected / 3 selected
+
+   eth2spec/test/merge/sanity/test_blocks.py .                              [ 33%]
+   eth2spec/test/phase0/sanity/test_blocks.py ..                            [100%]
+
+   =============================== warnings summary ===============================
+   ../../../venv/lib/python3.9/site-packages/cytoolz/compatibility.py:2
+     /home/qbzzt1/consensus-specs/venv/lib/python3.9/site-packages/cytoolz/compatibility.py:2: 
+   DeprecationWarning: The toolz.compatibility module is no longer needed in Python 3 and has 
+   been deprecated. Please import these utilities directly from the standard library. This 
+   module will be removed in a future release.
+       warnings.warn("The toolz.compatibility module is no longer "
+
+   -- Docs: https://docs.pytest.org/en/stable/warnings.html
+   ================ 3 passed, 626 deselected, 1 warning in 16.81s =================   
+   ```
+
+
+## The "Hello, World" of Consensus Spec Tests
+
+One of the `test_empty_block_transition` tests is implemented by a function with the same
+name located in 
+[`~/consensus-specs/tests/core/pyspec/eth2spec/test/phase0/sanity/test_blocks.py`](https://github.com/ethereum/consensus-specs/blob/dev/tests/core/pyspec/eth2spec/test/phase0/sanity/test_blocks.py).
+To learn how consensus spec tests are written, let's go over the code:
+
+```python
+@with_all_phases
+```
+
+This [decorator](https://book.pythontips.com/en/latest/decorators.html) specifies that this test
+is applicable to all the phases of the ETH 2.0 project. These phases are similar to forks (Istanbul,
```

Comment:
We are trying to deprecate the Eth2 nomenclature, in favor of "ethereum consensus layer"